Add content "requires user registration" component

B77Mills opened the pull request:
This will only account for required True/False, startDate & endDate.  
@Todo: If needed add support to account for [siteId] to allow for site specific gating.  Will need to decide if this will handle its own dates pre site.

https://github.com/parameter1/base-cms/pull/226

<img width="969" alt="Screen Shot 2022-01-26 at 11 39 16 AM" src="https://user-images.githubusercontent.com/3845869/151246439-2e27bc59-04ed-41d9-b7b0-db02dc86b2e3.png">
<img width="973" alt="Screen Shot 2022-01-26 at 11 48 48 AM" src="https://user-images.githubusercontent.com/3845869/151246443-8a180dc9-83a7-4c9b-9aef-a12f529e7d88.png">

B77Mills committed:
move buttons from scheduling to common folder

update refs in scheduling comonents too

B77Mills committed:
add fragment and mutation for userRegistration

B77Mills committed:
add new registration component

B77Mills committed:
add method for reseting of this value to a specific date

B77Mills committed:
cleanup and make cancel work correctly

update to handle when clickiing cancel to reset to last saved state.

B77Mills committed:
add registration component

B77Mills committed:
disable cancel unless there are changes

B77Mills commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Content UserRegistration by date management components](https://trello.com/c/IrcXjRpf/10-content-userregistration-by-date-management-components)

brandonbk reviewed (dismissed):
<img width="338" alt="Screen Shot 2022-01-26 at 5 05 36 PM" src="https://user-images.githubusercontent.com/2855198/151262070-5b7aea4c-ce08-4acc-bc3f-83a603a3b52b.png">

@B77Mills - could you adjust the spacing of the buttons to match the website scheduling?  So they aren't touching the inputs above?

B77Mills committed:
Fix classes and css for components and buttons

B77Mills committed:
update user reg fragment & mutation

B77Mills committed:
update registration component

 - udpate css to better match scheduling component
 - update fragment inclusion
 - update date disable logic

B77Mills committed:
Add enableSave & emit modelChanged

This will allow you to disable saving from the compoent and utilize the emit of userRegistrationChange with and { isRequired, startDate, endDate}

B77Mills committed:
change enableSave to saveEnabled & remove console.log

B77Mills committed:
No need to return this

B77Mills committed:
change emit from userRegistrationChange to just change
